### ISSUE TITLE
RF module upgrade discussion

### ISSUE THREAD

**Issue by csurf** [openLRSng/openLRSng#168]
Let's get the conversation started here on topics related to upgrading to a newer, better RF module...

I just recently received a couple of HopeRF RF69W 433 LoRa modules. Anyone have any experience with these? These are driven by the semtech SX1231 RF IC.

My goal is to figure out how to do FSK with them, but there isn't much info available on this yet outside of datasheet. Most people use these modules for LoRa modulation, so there's lots of info on how to do LoRa but not so much on the other modulation schemes.

**Comment by fiveangle** [comment]
Yes, it's quite a bad datasheet (almost as bad as the SX1278) but the info is there on p63 in the register table under _RegDataModul_ http://www.hoperf.com/upload/rfchip/RF69-V1.2.pdf

**Comment by mikerc2017** [comment]
Lora (i.e. Crossfire and any other LRS based on Lora) doesn't work properly when there is interference.
https://www.rcgroups.com/forums/showthread.php?2037442-Ultimate-LRS-40-km-APM-Telemetry-RC-link-for-54-%28using-2-OrangeRX-1W-TX%29/page850

**Comment by fiveangle** [comment]
https://www.rcgroups.com/forums/showpost.php?p=39048980&postcount=12755

**Comment by atomiclama** [comment]
Lots of reading to do. Thanks for all the links and info.

**Comment by atomiclama** [comment]
Another link for the pot of information. This one for a library to use the module. May be able to get some information as to how to configure it.

https://github.com/LowPowerLab/RFM69

**Comment by Linjieqiang** [comment]
Maybe we can use STM32F0 and LORA RX1279 to make this project again.

**Comment by Linjieqiang** [comment]
Maybe we can use STM32F0 and LORA RX1279 to make this project again.

**Comment by rodizio1** [comment]
Lora doesn't really work well when there is interference (see all the threads about the crossfire for example).

The Si4331 is completely outdated, since the IoT hype, manufacturers have improved sensitivity and interference-immunity quite a lot.

I'd either use something with a TI CC1310 or with Silabs si4464/63/62/61, these are much newer and better chips.

TI PDF comparing Lora with narrowband:
http://www.akermann.bg/files/Semnrs/H2016/Module3.4_100km%20TI%20NarrowBand%20vs%20LORA.pdf

Interference test against Lora:
http://www.youtube.com/watch?v=7zLUYswu3Ek

**Comment by csurf** [comment]
@rodizio1 
good info...

FYI, we don't want to necessarily use LoRa modulation. We'd still use FSK for RC control/telemetry. The consideration of LoRa modules is due to the fact that they use more modern RF IC's that can also do FSK (in order to move away from RFM). So, even though we're considering moving to LoRa-capable modules, we wouldn't necessarily modulate with LoRa.

**Comment by rodizio1** [comment]
Well, LoRa's "main feature" is the wideband Lora modulation, not sure how good the narrowband modulation modes work on them compared to TI's or Silabs' modules which are optimized/specialized for narrowband.

**Comment by csurf** [comment]
> not sure how good the narrowband modulation modes work on them compared to TI's or Silabs' modules which are optimized/specialized for narrowband.

This is a big question which we are trying to ascertain. The consensus is that we definitely want/need to move to a newer RF module, preferably something with better sensitivity and improved datarates if possible. The LoRa-capable modules seemed to be the most popular options, and it has been stated that both TBS and FrSky use them (I can't personally confirm this, but this is what other have claimed.

**Comment by mikeller** [comment]
I was just able to confirm for FrSky:

![img_20180130_010842](https://user-images.githubusercontent.com/4742747/35509872-c90c7fb6-055a-11e8-866a-7eee5716ac22.jpg)

**Comment by rodizio1** [comment]
Crossfire is also using Semtech, not sure which exact chip from the top of my head. It was developed by a swiss company (http://www.miromico.ch/) for TBS.

**Comment by jnb81** [comment]
what about SI4463? with -126dBm sensitivity
rx should gain 50% in reception comparing with current hw.

**Comment by MicroMontis** [comment]
I vote for SI4463 module. It should be pretty easy to implement support for it. 
Moreover there is confirmation that SI4432(RFM22B) should be able to transmit/receive packets to/from 
SI4463 (compatibility):
https://www.silabs.com/community/wireless/bluetooth/forum.topic.html/si4432_and_si4463pa-K9Tk
In this case we could mix the modules like RFM23 for TX and SI4463 for RX.

**Comment by Timvrakas** [comment]
I third the Si4463, and I'm doing hardware loss tests with Si4463 + TCXO prototypes I can share if anyone wants. Sticking with the SiLabs SPI interface vastly simplifies code changes, as the RFM22 was just an older chip of the same family.

**Comment by Rusty105** [comment]
@jnb81 

Assuming the noise floor is as low.

**Comment by dothien113** [comment]
video test SI4432  500mw TX and 100mw RX , it cheap 

https://www.youtube.com/watch?v=yw4sfvSYF9k&feature=youtu.be

**Comment by JamesCordell** [comment]
@csurf Is the LoRa CSS modulation not appropriate for rc controllers as opposed to FSK? Or has this not yet been studied.

**Comment by JamesCordell** [comment]
I have found the answer to my question. http://www.akermann.bg/files/Semnrs/H2016/Module3.4_100km%20TI%20NarrowBand%20vs%20LORA.pdf

Would the rfm98w and rmf98pw be a good candidate for a rx/tx setup?
